src/components/ColorDropdown/ColorDropdown.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Dropdown } from 'carbon-components-react';
+import {
+  purple70,
+  cyan50,
+  teal70,
+  magenta70,
+  red50,
+  red90,
+  green60,
+  blue80,
+  magenta50,
+  purple50,
+  teal50,
+  cyan90,
+} from '@carbon/colors';
+
+import { settings } from '../../constants/Settings';
+
+const { iotPrefix } = settings;
+
+const colorPropType = PropTypes.shape({
+  carbonColor: PropTypes.string,
+  name: PropTypes.string,
+});
+
+const propTypes = {
+  colors: PropTypes.arrayOf(colorPropType),
+  /** Internationalisation strings */
+  label: PropTypes.string,
+  titleText: PropTypes.string,
+  id: PropTypes.string.isRequired,
+  light: PropTypes.bool,
+  /** Callback for when any of the Dropdown color value changes */
+  onChange: PropTypes.func.isRequired,
+  selectedColor: colorPropType,
+  testID: PropTypes.string,
+};
```

Comment:
These comments are parsed by storybook and populated into the prop tables. Each one is individual, so we should be specific.

```suggestion
const propTypes = {
  colors: PropTypes.arrayOf(colorPropType),
  /** The label applied to the dropdown for internationalisation */
  label: PropTypes.string,
  /** The titleText applied to the dropdown for internationalisation */
  titleText: PropTypes.string,
  id: PropTypes.string.isRequired,
  light: PropTypes.bool,
  /** Callback for when any of the Dropdown color value changes */
  onChange: PropTypes.func.isRequired,
  selectedColor: colorPropType,
  testID: PropTypes.string,
};
```

Reply:
Yes, of course. I was a bit too fast when externalising this one, I'll add proper comments to all properties.